template/template-toml-update

```diff
@@ -25,4 +25,4 @@ else
 	sed -e 's/^# MANAGED BY FIOCONFIG: //g' -i /var/sota/sota.toml
 fi
 
-systemctl restart aktualizr-lite
+systemctl restart @@SOTA_CLIENT@@
```

Comment:
I have no strong opinion on the below, so I invite @mike-sul and @doanac to comment on this.

1. A comment on a file renaming operation:

It'd be nice to have a template like `contrib/aktualizr-toml-update.in`, so that it is clear to external developers of `fioconfig` what handlers are supported, without grepping the `meta-lmp-base`.
I mean, this project needs to be self-contained, as it is often being built and run outside LmP e.g. for development.

2. A comment on the templating itself:

I'd prefer if the `SOTA_CLIENT` goes from environment variable, similarly to the `CONFIG_FILE`.
Although, I'd give it the default value like this `${SOTA_CLIENT:-aktualizr-lite}` (so that no env variable equals "aktualizr-lite").
A reason is the same as above: we need an ability to build and run `fioconfig` outside LmP.

There are several ways how this environment variable can get into here:
- through templating of the `interrnal/app.go` at https://github.com/foundriesio/fioconfig/blob/main/internal/app.go#L303 in bb-append before `go build` (this is what I like the least).
- leveraging the `/etc/default/fioconfig` possibility at https://github.com/foundriesio/meta-lmp/blob/main/meta-lmp-base/recipes-support/fioconfig/fioconfig/fioconfig.service#L7 (added by @munoz0raul recently).
- adding the SOTA_CLIENT environment variable to `fioconfig/fioconfig.service` in bb-append (similar to above).

The latter two options make both `fioconfig` and `meta-lmp/.../fioconfig` fully encapsulated, but still provide the necessary level of integration and flexibilty.

Reply:
changing the value the systemd service name from immutable (ostree) to an environment variable may have security ramifications. This templating approach might be more "secure".

Reply:
The original version that Caio posted was kind of "self-contained" (a new Makefile target that turns the template to actual value).

I don't like this approach because the update script customization is Yocto/meta-lmp specific, it's not needed for the fioconfig project at all.

Relying on the env variable is not a good option because the sota client is defined at build time, so the toml update script should be defined at build time and it's behavior should not be customizable at runtime. Otherwise, customers can easily shoot themself into the foot.

Ideally, I would make this customization (which of the custom client to restart in the `aktualizr-toml-update`) solely within `meta-lmp` no any changes to fioconfig.

Maybe, we can make the fioconfig recipe replace `systemctl restart aktualizr-lite` with `systemctl restart ${SOTA_CLIENT}` so no need to make any changes to fioconfig project at all?

Reply:
I would do in `do_configure:append()`

```
sed -e 's|systemctl restart aktualizr-lite|systemctl restart ${SOTA_CLIENT}|g' \
        ${S}/src/${GO_IMPORT}/template/template-toml-update > ${S}/src/${GO_IMPORT}/template/template-toml-update-customized
```

And then in `do_install:append()`

```
install -m 0755 ${S}/src/${GO_IMPORT}/contrib/aktualizr-toml-update-customized ${D}${datadir}/fioconfig/handlers
```

Since the fioconfig is linked/bound with specific version of `fioconfig` (`SRCREV=`) I don't see any issues with "sedding" of `systemctl restart aktualizr-lite`.

Reply:
@doanac @mike-sul I buy your arguments; that's why I told that I have no strong opinion.

Actually, I like the latest suggestion from @mike-sul, which is to replace the whole thing.
I _believe_ we can improve on it by declaring a local script variable:
```
SOTA_CLIENT=aktualizr-lite

systemctl restart ${SOTA_CLIENT}
```

This way, the bb-append script can do a more intuitive substitution:
```
sed -e "s|SOTA_CLIENT=aktualizr-lite|SOTA_CLIENT=${SOTA_CLIENT}|g" -i ${S}/src/${GO_IMPORT}/template/template-toml-update > > ${S}/src/${GO_IMPORT}/template/template-toml-update-customized
```
It is also forward-compatible, as it allows to reuse the SOTA_CLIENT in all _future_ places inside a script where we need it.

Reply:
The improvement makes sense to me.

Reply:
@mike-sul , nice, this is more simple.